extensions/interactions/GraphInput/directives/graph-viz.component.ts

```diff
@@ -277,15 +323,51 @@ export class GraphVizComponent implements OnInit, AfterViewInit {
     }
   }
 
+  isButtonOnTopOfDot(): boolean {
+    const dotElement = document.querySelector(
+      '.oppia-add-node-cursor') as HTMLDivElement;
+    const buttonElements = document.querySelectorAll(
+      '[class^="e2e-test-"][class$="-button"]');
```

Comment:
What exactly are the buttonElements in this context?

Note that the selectors look too generic. "e2e-test-" is used throughout the codebase for e2e tests to find/interact with elements.

Reply:
@kevintab95  The buttonElements are the buttons of the graph "Move", "Add Node", "Add edge", "Delete" that are on the graph. This is only done to keep track of the user clicking on the buttons of the graph in the case when the dot cursor is shown to the user.
![Screenshot 2023-07-15 140502](https://github.com/oppia/oppia/assets/96219910/f61c983a-3e04-4e29-aa68-7d79deca1264)

Reply:
Please see my point regarding using "e2e-test-*" classes in the code. Can we either use a different class that isn't specific to e2e tests or define new ones for this purpose?

Reply:
Hi @kevintab95 I have added new the classname PTAL. Thanks!